File: Web/src/core/auth/tokenStorage.ts
/** Persists the access/refresh token pair in localStorage across page loads.
 * A browser SPA has no Keychain/Keystore equivalent; localStorage is the
 * pragmatic choice given the backend only supports Bearer tokens (no
 * httpOnly-cookie session support). */
const ACCESS_TOKEN_KEY = 'diveapp.accessToken';
const REFRESH_TOKEN_KEY = 'diveapp.refreshToken';

export const tokenStorage = {
  get accessToken(): string | null {
    return localStorage.getItem(ACCESS_TOKEN_KEY);
  },
  set accessToken(value: string | null) {
    if (value) localStorage.setItem(ACCESS_TOKEN_KEY, value);
    else localStorage.removeItem(ACCESS_TOKEN_KEY);
  },
  get refreshToken(): string | null {
    return localStorage.getItem(REFRESH_TOKEN_KEY);
  },
  set refreshToken(value: string | null) {
    if (value) localStorage.setItem(REFRESH_TOKEN_KEY, value);
    else localStorage.removeItem(REFRESH_TOKEN_KEY);
  },
  save(accessToken: string, refreshToken: string) {
    this.accessToken = accessToken;
    this.refreshToken = refreshToken;
  },
  clear() {
    this.accessToken = null;
    this.refreshToken = null;
  },
};

File: Frontend/DiveApp/Shared/Auth/TokenStorage.swift
import Foundation

/// Persists the access/refresh token pair in the Keychain across app launches.
final class TokenStorage {
    private enum Key {
        static let accessToken = "com.diveapp.accessToken"
        static let refreshToken = "com.diveapp.refreshToken"
    }

    var accessToken: String? {
        get { KeychainHelper.read(forKey: Key.accessToken) }
        set {
            if let newValue {
                KeychainHelper.save(newValue, forKey: Key.accessToken)
            } else {
                KeychainHelper.delete(forKey: Key.accessToken)
            }
        }
    }

    var refreshToken: String? {
        get { KeychainHelper.read(forKey: Key.refreshToken) }
        set {
            if let newValue {
                KeychainHelper.save(newValue, forKey: Key.refreshToken)
            } else {
                KeychainHelper.delete(forKey: Key.refreshToken)
            }
        }
    }

    func save(accessToken: String, refreshToken: String) {
        self.accessToken = accessToken
        self.refreshToken = refreshToken
    }

    func clear() {
        accessToken = nil
        refreshToken = nil
    }
}
